index.js

```diff
@@ -1,88 +1,116 @@
 
-// console.log(drinkWater(12))
-// // => "Man I sure am thirsty"
-// // => "Ahh that hits the spot"
-// // => 11
-// console.log("%c----------", "color: red") 
+function drink_water(current_thirst_level) {
+  console.log("Man I sure am thirsty")
+  current_thirst_level -= 1 
+  console.log("Ahh that hits the spot")
+  return current_thirst_level
+}
+console.log(drinkWater(12))
+// => "Man I sure am thirsty"
+// => "Ahh that hits the spot"
+// => 11
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
-
-// console.log(sameSameButDifferent(5, 5)) 
-// // => "same same"
+console.log("%cQuestion 4", "color: red") 
+
+function sameSameButDifferent(num, maybeNum) {
+  if num === maybeNum 
+    return "same same"
+  else if num == maybeNum 
+    return "same same (but different)"
+  else 
+  return "different"
+}
+console.log(sameSameButDifferent(5, 5)) 
+// => "same same"
 
-// console.log(sameSameButDifferent(123, "123")) 
-// // => "same same (but different)"
+console.log(sameSameButDifferent(123, "123")) 
+// => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 7)) 
-// // => "different"
+console.log(sameSameButDifferent(5, 7)) 
+// => "different"
 
-// console.log(sameSameButDifferent(123, "122")) 
-// // => "different"
-// console.log("%c----------", "color: red") 
+console.log(sameSameButDifferent(123, "122")) 
+// => "different"
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 5 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)
-// // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red")
+
+function updateGrade(student, grade) {
+  student['grade'] = grade 
+}
+console.log("%cQuestion 5", "color: red")
+const student1 = { name: "Duane", grade: 88 }
+updateGrade(student1, 92)
+console.log(student1)
+// => { name: "Duane", grade: 92 }
+console.log("%c----------", "color: red")
 
 
 
 // ***** Question 6 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 6", "color: red")
-
-// const users = [ 
-//   { 
-//     name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
-//   },
-//   { 
-//     name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
-//   }
-// ]
-// printNameAndPhones(users)
-// // => "Duane"
-// => "Cell: 555-123-4567"
-// => "Office: 555-456-7890"
-// => "Liza"
-// => "Cell: 555-234-5678"
-// => "Office: 555-567-1234"
-// console.log("%c----------", "color: red") 
+console.log("%cQuestion 6", "color: red")
+
+function printNameAndPhones(user) {
+  for (let i = 0; i < user.length; i++)
+    console.log([i]['name'])
+    console.log(`Cell: ${[i]['phones']['cell']}`)
+    console.log(`Office: ${[i]['phones']['office']}`)
+  }
+}
```

Comment:
close! If you want to use the for loop, it should look like this:

```javascript
for (let i = 0; i < user.length; i++) {
    console.log(user[i]['name'])
    console.log(`Cell: ${user[i]['phones']['cell']}`)
    console.log(`Office: ${user[i]['phones']['office']}`)
}
```

- You can also use dot notation instead of bracket notation:
```javascript
for (let i = 0; i < user.length; i++) {
    console.log(user[i].name)
    console.log(`Cell: ${user[i].phones.cell`)
    console.log(`Office: ${user[i].phones.office`)
}
```
____

For iterating over arrays, Javascript has a couple other methods that make it a little easier.
- Here's how it would look with for..of:
```javascript
  for (let user of users) {
    console.log(user.name);
    console.log(`Cell: ${user.phones.cell}`);
    console.log(`Office: ${user.phones.office}`);
  }
```
- Here's how it would look with forEach:
```javascript
  users.forEach(function(user) {
    console.log(user.name);
    console.log(`Cell: ${user.phones.cell}`);
    console.log(`Office: ${user.phones.office}`);
  })
```